force-app/main/default/classes/PatientRelatedCasesTest.cls

```diff
@@ -21,6 +21,6 @@ public class PatientRelatedCasesTest {
         List<Case> relatedCases = PatientRelatedCases.query(case1.Id);
         Test.stopTest();
 
-        system.assertEquals(2, relatedCases.size());   
+        system.assertEquals(1, relatedCases.size());   
```

Comment:
@deepakmulamalla : I can see the lines 5 to 18 above this change will create the record that are required to pass the assertion. This will work in the context of System admin because he will have access to Drug, Account and Case. Like i mentioned before, you should be checking for negative assertions too. Also i noticed there is no assertion message in System.assertEquals, please update the same and add negative assertion and message too.

Reply:
Changes have been made accordingly to PatientRelatedCasesTest class